lib/remote/apilistener.cpp

```diff
@@ -368,7 +368,41 @@ void ApiListener::Stop(bool runtimeDeleted)
 	m_Timer->Stop(true);
 	m_RenewOwnCertTimer->Stop(true);
 
+	m_ListenerWaitGroup->Cancel();
+	m_ListenerWaitGroup->Join();
+
+	DisconnectJsonRpcConnections();
+
+	{
+		std::unique_lock lock(m_HttpClientsLock);
+		Log(LogInformation, "ApiListener")
+			<< m_HttpClients.size() << " HTTP connections remaining.";
+	}
+
+	m_Timer->OnTimerExpired.disconnect_all_slots();
+	m_Timer->OnTimerExpired.connect([this](const Timer *const&) {
+		m_WaitGroup->Cancel();
+		DisconnectHttpClients();
+		m_Timer->Stop();
+	});
+	m_Timer->SetInterval(3);
+	m_Timer->Start();
```

Comment:
I'm not the biggest fan of recycling `m_Timer` here. That name already isn't very helpful and this additional usage won't make it better for anyone trying to figure out what it does.